tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -31,44 +32,56 @@ private[scalanative] object LLVM {
    *  @return
    *    The paths of the `.o` files.
    */
-  def compile(config: Config, paths: Seq[Path]): Seq[Path] = {
+  def compile(config: Config, paths: Seq[Path])(implicit
+      incCompilationContext: IncCompilationContext = null
+  ): Seq[Path] = {
     // generate .o files for all included source files in parallel
     paths.par.map { path =>
       val inpath = path.abs
       val outpath = inpath + oExt
       val isCpp = inpath.endsWith(cppExt)
       val isLl = inpath.endsWith(llExt)
       val objPath = Paths.get(outpath)
+      val packageName = (config.workdir relativize path).toString
+        .replace(File.separator, ".")
+        .split("\\.")
+        .dropRight(1)
+        .mkString(".")
```

Comment:
I think we can make it more stable by usage of `Path.subPath`, it might be more safe then parsing string path, here's example, which could be simpler if we were allowed to use collection conventers here (it's not possible due to cross compilation with Scala 2.12) 
```suggestion
      val packageName = { 
       val relPath = (config.workdir relativize path)
       val it = relPath.subPath(0, relPath.getNameCount() - 1).iterator
       // conversion from java to scala iterator
       new Iterator[Path]{def hasNext() = it.hasNext(); def next() = it.next()}
         .mkString(".")
```